driver-core/src/test/unit/com/mongodb/internal/connection/TestServerListener.java

```diff
@@ -23,31 +23,47 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Predicate;
 
-import static com.mongodb.assertions.Assertions.isTrue;
 import static com.mongodb.assertions.Assertions.notNull;
 
-class TestServerListener implements ServerListener {
+public class TestServerListener implements ServerListener {
```

Comment:
The changes in this file match roughly what's in TestConnectionPoolListener, which lets us wait for events efficiently.

Reply:
Ack